src/Microsoft.AspNet.Mvc.TagHelpers/LinkTagHelper.cs

```diff
@@ -229,15 +229,19 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
 
             var builder = new DefaultTagHelperContent();
 
-            if (mode == Mode.Fallback && string.IsNullOrEmpty(HrefInclude) || mode == Mode.AppendVersion)
+            if (!string.IsNullOrEmpty(Href))
```

Comment:
This condition seems incorrect since the previous code would have added the file version to any `href` attribute in the `output` as long as `AppendVersion` was `true`. (Another negative aspect of the `AppendFileVersionIfApplicable()` refactoring is separating this condition from those in the new method.)

In any case handle the `href` attribute in the original element immediately after restoring it and bring the conditions back together, removing `AppendFileVersionIfApplicable()`.

Separately, the previous version of this class did not rely on `Href` in this fashion, allowing earlier tag helpers to add an `href` attribute to the `output` without strange side-effects here.